sdk/core/azure-core-amqp/inc/azure/core/amqp/session.hpp

```diff
@@ -5,161 +5,154 @@
-          Azure::Core::Amqp::Models::AmqpValue target,
-          Azure::Core::Amqp::Models::AmqpValue properties)
-          = 0;
-    };
-
-    class Session final {
-    public:
-      Session(
-          Connection const& parentConnection,
-          Endpoint& newEndpoint,
-          SessionEvents* eventHandler = nullptr);
-      Session(Connection const& parentConnection, SessionEvents* eventHandler = nullptr);
-      Session(std::shared_ptr<Azure::Core::Amqp::_detail::SessionImpl> impl) : m_impl{impl} {}
-      ~Session() noexcept;
-
-      Session(Session const&) = delete;
-      Session& operator=(Session const&) = delete;
-      Session(Session&&) noexcept = delete;
-      Session& operator=(Session&&) noexcept = delete;
-      std::shared_ptr<Azure::Core::Amqp::_detail::SessionImpl> GetImpl() const { return m_impl; }
-      void SetIncomingWindow(uint32_t incomingWindow);
-      uint32_t GetIncomingWindow() const;
-      void SetOutgoingWindow(uint32_t outgoingWindow);
-      uint32_t GetOutgoingWindow() const;
-      void SetHandleMax(uint32_t handleMax);
-      uint32_t GetHandleMax() const;
-
-      void Begin();
-      void End(std::string const& condition_value, std::string const& description);
-
-    private:
-      std::shared_ptr<Azure::Core::Amqp::_detail::SessionImpl> m_impl;
-    };
-  } // namespace _internal
-}}} // namespace Azure::Core::Amqp
+namespace Azure { namespace Core { namespace Amqp { namespace _detail {
+  class SessionImpl;
+  class SessionFactory;
+}}}} // namespace Azure::Core::Amqp::_detail
+
+namespace Azure { namespace Core { namespace Amqp { namespace _internal {
+
+  class Connection;
+  enum class SessionRole;
+
+  enum class ExpiryPolicy
+  {
+    LinkDetach,
+    SessionEnd,
+    ConnectionClose,
+    Never
+  };
+
+  enum class SessionState
+  {
+    Unmapped,
+    BeginSent,
+    BeginReceived,
+    Mapped,
+    EndSent,
+    EndReceived,
+    Discarding,
+    Error,
+  };
+
+  enum class SessionSendTransferResult
+  {
+    Ok,
+    Error,
+    Busy,
+  };
+
+  class Session;
+  struct SessionEvents
+  {
+    virtual bool OnLinkAttached(
+        Session const& session,
+        LinkEndpoint& newLink,
+        std::string const& name,
+        SessionRole role,
+        Azure::Core::Amqp::Models::AmqpValue source,
+        Azure::Core::Amqp::Models::AmqpValue target,
+        Azure::Core::Amqp::Models::AmqpValue properties)
+        = 0;
+  };
+
+  struct SessionOptions final
+  {
+    /** @brief The Audience to which an authentication operation applies when using Claims Based
+     * Authentication. */
+    std::vector<std::string> AuthenticationScopes;
+
+    /** @brief Represents the initial incoming window size for the sender. See [AMQP Session Flow
+     * Control](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#doc-session-flow-control)
+     * for more information.*/
+    Azure::Nullable<uint32_t> InitialIncomingWindowSize;
```

Comment:
```suggestion
    Nullable<uint32_t> InitialIncomingWindowSize;
```

There are also similar instances below.